test/Microsoft.Azure.WebJobs.Host.UnitTests/AttributeClonerTests.cs

```diff
@@ -140,6 +141,35 @@ public string TemplateBind(PropertyInfo propInfo, Attribute attribute, BindingTe
             }
         }
 
+        // Validation with AutoResolve
+        [Binding]
+        public class ValidationWithAutoResolveAttribute : Attribute
+        {
+            [RegularExpression("a+")]
+            [AutoResolve]
+            public string Value { get; set; }
+
+        }
+
+        // Validation with AppSetting
+        [Binding]
+        public class ValidationWithAppSettingAttribute : Attribute
+        {
+            [RegularExpression("a+")]
+            [AppSetting]
+            public string Value { get; set; }
+        }
+
+        // Validation with neither AppSetting nor AutoResolve
+        [Binding]
+        public class ValidationOnlyAttribute : Attribute
+        {
+            // Allow  { } that look like token substitution, but it's not since this isn't AutoResolve
+            [RegularExpression("^.{1,3}$")] 
```

Comment:
We now support these declarative validation attributes for binding authors. Might be interesting to support them for end users as well on their model types. E.g. I mark up my queue trigger poco, and our pipeline runs the validation and fails the invoke if the validations don't pass. It'd be a new validation stage in our pipeline.